### PR TITLE
add EF core 7.0  support

### PR DESCRIPTION
**Pull request by MehdiHaghshenas**

### CODE REVIEW

File: EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj [hunk 1]
```diff
@@ -1,20 +1,20 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj:4]: EF Core 7 targets net 6

Comment (MehdiHaghshenas) [reply EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj:4]: For Complete Test of this project we should change target framework of Test Project to `net7.0` and use visual studio preview  to build

File: Directory.Build.props [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <Copyright>© 2022 Brice Lambson, et al. All rights reserved.</Copyright>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)key.snk</AssemblyOriginatorKeyFile>
-    <Version>3.0.1</Version>
+    <Version>3.0.2</Version>
```
Comment (ErikEJ) [inline Directory.Build.props:10]: I think it should be 4.0.0-rc.1

**Comment by ErikEJ** [comment]
Plans to publish a rc1/rc2 package?

**Comment by bricelam** [comment]
Published. I’ll try to release RC2 (and GA) the same day as EF.

**Pull request merged**